src/router/legacyPermalinkManagement.routerPlugin.js

```diff
@@ -57,8 +59,11 @@ const handleLegacyParams = (legacyParams, store, to, next) => {
             case 'zoom':
                 // if we are using WebMercator as the map projection we need te re-evaluate LV95 zoom,
                 // as it was a zoom level tailor-made for this projection (and not made to cover the whole globe)
-                if (store.state.position.projection.epsg === WEBMERCATOR.epsg) {
-                    value = translateSwisstopoPyramidZoomToMercatorZoom(legacyParams[param])
+                if (!(projection instanceof SwissCoordinateSystem)) {
+                    value = LV95.transformCustomZoomLevelToStandard(legacyParams[param])
+                    if (projection instanceof CustomCoordinateSystem) {
+                        value = projection.transformStandardZoomLevelToCustom(value)
+                    }
```

Comment:
Nice one but the comment needs to be updated. The code is almost self explaining and I think that just a comment that `legacyParams['zoom']` is always in LV95 would be sufficient.